lib/puppet/provider/ssl_pkey/openssl.rb

```diff
@@ -28,6 +28,7 @@ def self.to_pem(resource, key)
   end
 
   def exists?
+    return false if resource[:regenerate]
```

Comment:
I'll think on this. One of our primary use cases is the use of the Kafo project (github.com/theforeman/kafo) to turn a group of puppet modules into an installer for our users (https://github.com/Katello/katello-installer  -- in case you are curious. In some cases we may want to allow users to supply their own CA prior to or after an installation, or to provider their web/server certificates but use the CA we generate by default which would lead to a need to regenerate. Or you could think of the case where you wanted to generate new certs to invalidate the current ones.